src/pet.js

```diff
@@ -0,0 +1,80 @@
+
+const MAXIMUM_FITNESS = 10;
+const MINIMUM_HUNGER = 0;
+const MINIMUM_NEEDFEEDING = 5;
+const MINIMUM_NEEDWALK = 3;
+
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = MAXIMUM_FITNESS;
+    this.children = [];
+};
+
+Pet.prototype = {
+    get isAlive() {
+      return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    }
+  };
+
+Pet.prototype.growUp = function() {
+    if (!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+      } else {
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+      };
+};
+
+Pet.prototype.walk = function() {
+    if (!this.isAlive) {
+        throw new Error('Your pet is no longer alive :(');
+      } else if ((this.fitness + 4) <= MAXIMUM_FITNESS) {
+        this.fitness += 4;
+    } else {
+        this.fitness = MAXIMUM_FITNESS;
+    }
```

Comment:
This could be refactored a bit. In essence you are adding a value to the existing fitness, without allowing it go over a set limit. You can do this with Math.min, which will return the lowest of the values passed to it:
```
this.fitness = Math.min(this.fitness + 4, MAXIMUM_FITNESS)
```